FIX: Reset self._channels when updating channel property.

hhslepicka opened the pull request:
Attn. @gabrielfedel

hmbui commented on pydm/tests/widgets/test_base.py:
```diff
@@ -353,6 +353,32 @@ def test_channels_for_tools(qtbot):
+    pydm_label.channel = 'foo://bar'
+    assert pydm_label._channel == 'foo://bar'
+    assert pydm_label.channels()[0].address == 'foo://bar'
+    assert pydm_label._channels is not None
```
If the previous assertion is True, there's no need to assert pydm_label._channels is not None. Either remove this assertion, or assert for this before asserting the _channels[0].address.

hhslepicka commented:
Thank you, @hmbui ... I addressed your comment in the latest commit.
The situation here is:
- Create the widget
- From nothing, set the channel...
- From a channel, change it to another channel...
That is why the test looks repeated and redundant but it was a purpose. 👍
